### PR TITLE
Make Sequencer emit redstone signal when at end of tape

### LINKED ISSUE #25: [Suggestion] Some way to tell that a sequencer has looped or halted
There is currently no way to indicate that a sequencer loops or needs to be manually reset.

This could be implemented by making the face with the tape emit a redstone signal when the tape is at the end ( hold for manual reset mode and a pulse for automatic reset mode). Having that face emit the signal will prevent that signal interfering with the output of the sequencer, even in analog mode. 

Having the information that a sequencer resets could be used to string multiple sequencers together and get even more complex logic

### PR DESCRIPTION
**Pull request by keifhausin**
Closes #25

### CODE REVIEW

**Review by malte0811 — CHANGES_REQUESTED** [review]
Two additional points:
1. Please ask me before implementing features in the future (comment on the issue in question). In some cases I have my own "little ideas" about how a feature should be implemented.
2. Before I start debugging this in more detail, do you know why the Github actions didn't run on your commits? As far as I can tell `push` should have run on your commits even though they are in a fork.

**Comment by keifhausin** [comment]
Sorry for not asking in the ticket first, ill do that in future.
The Github actions didn't run in my repo on push is Github auto disables actions in forks, as for why it didn't run on the PR I think that's because PR creation uses a different action type. looks like it uses `pull_request` https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request.

File: src/main/java/malte0811/controlengineering/blockentity/tape/SequencerBlockEntity.java [hunk 1]
```diff
@@ -82,11 +82,17 @@ public void tick() {
...
             if (newPos < tape.getTapeLength()) {
+            	level.setBlockAndUpdate(worldPosition, blockState.setValue(SequencerBlock.HALTED, false));
                 currentTapePosition = newPos;
-            } else if (autoreset) {
-                currentTapePosition = 0;
+            } else {
+            	if (autoreset) {
+            		currentTapePosition = 0;
+            	}
+                level.setBlockAndUpdate(worldPosition, blockState.setValue(SequencerBlock.HALTED, true));
```
[the remark this answers is not in the document]
Comment (keifhausin) [reply src/main/java/malte0811/controlengineering/blockentity/tape/SequencerBlockEntity.java]: Fixed the tabs everywhere.

File: src/main/java/malte0811/controlengineering/blockentity/tape/SequencerBlockEntity.java [hunk 2]
```diff
@@ -82,11 +82,17 @@ public void tick() {
         final Direction clockFace = getBlockState().getValue(SequencerBlock.FACING).getCounterClockWise();
         boolean rsIn = level.getSignal(worldPosition.relative(clockFace), clockFace) > 0;
         if (clock.getClock().tick(rsIn) && !energy.extractOrTrue(CONSUMPTION_PER_STEP)) {
+        	BlockState blockState = getBlockState();
+        	
             int newPos = ++currentTapePosition;
             if (newPos < tape.getTapeLength()) {
+            	level.setBlockAndUpdate(worldPosition, blockState.setValue(SequencerBlock.HALTED, false));
```
[the remark this answers is not in the document]
Comment (keifhausin) [reply src/main/java/malte0811/controlengineering/blockentity/tape/SequencerBlockEntity.java]: Added checks to only change it if it needs to be changed

File: src/main/java/malte0811/controlengineering/blocks/tape/SequencerBlock.java [hunk 3]
```diff
@@ -48,4 +54,20 @@ public <T extends BlockEntity> BlockEntityTicker<T> getTicker(
         }
         return super.getTicker(pLevel, pState, pBlockEntityType);
     }
+    
+    @Override
+    public boolean canConnectRedstone(BlockState state, BlockGetter level, BlockPos pos, Direction direction) {
+    	Direction facing = state.getValue(FACING);
+    	return facing != null &&( facing == direction || facing == direction.getCounterClockWise() );
```
[the remark this answers is not in the document]
Comment (keifhausin) [reply src/main/java/malte0811/controlengineering/blocks/tape/SequencerBlock.java]: removed the null checks and fixed the formatting

File: src/main/java/malte0811/controlengineering/blocks/logic/LogicCabinetBlock.java [hunk 4]
```diff
@@ -75,4 +77,11 @@ public <T extends BlockEntity> BlockEntityTicker<T> getTicker(
     public static boolean isMaster(BlockState state) {
         return state.getValue(HEIGHT) == 0;
     }
+    
+    @Override
+    public boolean canConnectRedstone(BlockState state, BlockGetter level, BlockPos pos, Direction direction) {
+    	if (direction == null) {return false;};
+    	Direction facing = state.getValue(FACING);
+    	return isMaster(state) && facing != null && facing == direction.getCounterClockWise();
```
[the remark this answers is not in the document]
Comment (keifhausin) [reply src/main/java/malte0811/controlengineering/blocks/logic/LogicCabinetBlock.java]: I added the equivalent function to the LogicCabinetBlock and used that.  I removed the `leftIfNonMirrored` parameter from the block version as it felt redundant to hard code a `true` in the only call to that function.

**Comment by malte0811** [comment]
> Github auto disables actions in forks

Good to know, I was quite sure I had used actions in forks before… Maybe it changed at some point, or I'd enabled some option on those forks that I forgot about. I'll add the PR trigger as well.

Thanks for implementing this!

**Pull request merged**